components/standings/commons/standings.lua

```diff
@@ -0,0 +1,172 @@
+
+local OpponentLibraries = require('Module:OpponentLibraries')
+local Opponent = OpponentLibraries.Opponent
+
+local Standings = {}
+
+---@class StandingsModel
+---@field pageName string
+---@field standingsIndex integer
+---@field tournament StandardTournament?
+---@field title string?
+---@field section string?
+---@field type 'ffa'|'swiss'|'league'
+---@field matches MatchGroupUtilMatch[]
+---@field config table
+---@field rounds StandingsRound[]
+---@field private lpdbdata standingstable
+
+---@class StandingsRound
+---@field round integer
+---@field finished boolean
+---@field title string
+---@field opponents StandingsEntryModel[]
+
+---@class StandingsEntryModel
+---@field opponent standardOpponent
+---@field placement string
+---@field position integer
+---@field points number
+---@field positionStatus string?
+---@field definitiveStatus string?
+---@field positionChangeFromPreviousRound integer
+---@field pointsChangeFromPreviousRound number
+
+---@param pagename string
+---@param standingsIndex integer #0-index'd on per page
+---@return StandingsModel?
+function Standings.getStandingsTable(pagename, standingsIndex)
+	local pageNameInCorrectFormat = string.gsub(pagename, ' ', '_')
+	local record = mw.ext.LiquipediaDB.lpdb('standingstable', {
+		conditions = '[[pagename::' .. pageNameInCorrectFormat .. ']] AND [[standingsindex::' .. standingsIndex .. ']]',
+		limit = 1,
+	})[1]
+	if not record then
+		return nil
+	end
+	return Standings.standingsFromRecord(record)
+end
+
+local StandingsMT = {
+	__index = function(standings, property)
+		if property == 'tournament' then
+			standings[property] = Tournament.getTournament(standings.pageName)
+		end
+		if property == 'matches' then
+			standings[property] = Standings.fetchMatches(standings)
+		end
+		if property == 'rounds' then
+			standings[property] = Standings.makeRounds(standings)
+		end
+		return rawget(standings, property)
+	end
+}
+
+---@param record standingstable
+---@return StandingsModel
+function Standings.standingsFromRecord(record)
+	local standings = {
+		pageName = record.pagename,
+		standingsIndex = record.standingsindex,
+		title = record.title,
+		section = record.section,
+		type = record.type,
+		config = record.config,
+		rounds = record.rounds,
+		lpdbdata = record,
+	}
+
+	-- Some properties are derived from other properies and we can calculate them when accessed.
+	setmetatable(standings, StandingsMT)
+
+	return standings
+end
+
+---@param record standingsentry
+---@return StandingsEntryModel
+function Standings.entryFromRecord(record)
+	local entry = {
+		opponent = Opponent.fromLpdbStruct(record),
+		placement = record.placement,
+		position = tonumber(record.slotindex),
+		positionStatus = record.currentstatus,
+		definitiveStatus = record.definitestatus,
+		points = record.scoreboard.points,
+		pointsChangeFromPreviousRound = record.extradata.pointschange,
+		positionChangeFromPreviousRound = tonumber(record.placementchange),
+	}
+
+	return entry
+end
```

Comment:
what about the other values?
- rest of scoreboard (will be needed for gtl/stl!)
- extradata (will be needed for gtl/stl)

(maybe oos for this use case)

Reply:
out of scope for this, more will be added when it's needed